docs/developer/design/dashboard-architecture.md

```diff
@@ -0,0 +1,660 @@
+flowchart LR
+    Model["Model<br/>(ConfigService, DataService)"]
+    Controller["Controller<br/>(WorkflowController, BaseParamModel)"]
+    View["View<br/>(Panel Widgets)"]
+
+    View <--> Controller
+    Controller <--> Model
+```
+
+### Subscription and Notification Flow
+
+```mermaid
+sequenceDiagram
+    participant ConfigService
+    participant DataService
+    participant WorkflowController
+    participant BaseParamModel
+    participant PanelWidgets
+
+    ConfigService->>WorkflowController: Notify config update
+    DataService->>WorkflowController: Notify data update
+    WorkflowController->>PanelWidgets: Notify relevant update
+    PanelWidgets->>WorkflowController: User actions (start/stop workflow)
+    WorkflowController->>ConfigService: Update config
+    WorkflowController->>DataService: Update data (e.g., cleanup)
+
+    ConfigService->>BaseParamModel: Notify config update (simple controls)
+    BaseParamModel->>PanelWidgets: Notify widget update
+    PanelWidgets->>BaseParamModel: User changes widget (simple controls)
+    BaseParamModel->>ConfigService: Update config (simple controls)
+```
+
+### Analysis
+
+**Strengths:**
+- Clear separation of concerns with well-defined responsibilities
+- Testability through controller and widget isolation using fakes
+- Maintainability with centralized business logic
+- Extensibility for future requirements
+- Efficient handling of simple controls via `BaseParamModel`
+
+**Potential Pitfalls:**
+- Controller bloat as it mediates more services
+- State synchronization challenges between controller and services
+- Subscription complexity with multiple layers
+
+**Anti-Patterns Avoided:**
+- No leaky abstractions between views and services
+- No tight coupling between controllers and specific GUI frameworks
+- No direct service access from views
+
+## Background Threading Architecture
+
+### KafkaBridge Threading Model
+
+```mermaid
+sequenceDiagram
+    participant GUI as GUI Thread
+    participant KB as KafkaBridge
+    participant BT as Background Thread
+    participant K as Kafka
+
+    Note over GUI,K: Startup
+    GUI->>KB: start()
+    KB->>BT: spawn thread
+    BT->>K: subscribe to topic
+
+    Note over GUI,K: Publishing (Non-blocking)
+    GUI->>KB: publish(key, value)
+    KB->>KB: queue.put()
+    BT->>KB: queue.get()
+    BT->>K: produce message
+
+    Note over GUI,K: Consuming (Batched)
+    K->>BT: poll messages
+    BT->>KB: incoming_queue.put()
+    GUI->>KB: process_incoming_messages()
+    KB->>GUI: pop_message() × N
+```
+
+### Message Processing Strategy
+
+The KafkaBridge implements several optimizations:
+
+1. **Batched Processing**: Consumes up to `max_batch_size` messages per poll
+2. **Timed Polling**: Only checks for incoming messages at specified intervals
+3. **Queue-based Communication**: Non-blocking queues between GUI and background threads
+4. **Smart Idle Handling**: Minimal CPU usage when no messages are available
+
+## Key Architectural Principles
```

Comment:
The subsections in this section either duplicate info from other sections or do not relate to anything else in this document. Can this just be removed?